lib/location/location_utils.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdio.h>
+#include <zephyr.h>
+#include <logging/log.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <nrf_modem_at.h>
+#include <modem/at_cmd_parser.h>
+#include <modem/at_params.h>
+#include <net/nrf_cloud.h>
+
+#include "location_utils.h"
+
+LOG_MODULE_DECLARE(location, CONFIG_LOCATION_LOG_LEVEL);
+
+#define AT_CMD_PDP_ACT_READ "AT+CGACT?"
+
+char jwt_buf[600];
+
+bool location_utils_is_default_pdn_active(void)
+{
+	char at_response_str[128];
+	const char *p;
+	int err;
+	bool is_active = false;
+
+	err = nrf_modem_at_cmd(at_response_str, sizeof(at_response_str), "%s", AT_CMD_PDP_ACT_READ);
```

Comment:
Not sure how big is this response, but you can scanf `%128s` which will make sure there are no overruns.

Reply:
Not sure what would be the best method to use here. I'd assume `nrf_modem_at_cmd` would be lighter than `nrf_modem_scanf`.

Reply:
If you what to copy the entire response, `nrf_modem_at_cmd` is probably better since it does a simple `memcpy` which is lighter than a `sscanf`. If you want to parse multiple values from the response, with `nrf_modem_at_scanf` you avoid any intermediate copies and can send the command and parse the response in one function call.

Anyhow, here you don't need to format the command `nrf_modem_at_cmd(buf, size, AT_CMD_PDP_ACT_READ)`.

Reply:
Yep, this works without formatting so changed.